musysim.py

```diff
@@ -78,27 +157,13 @@ def store_input(self, input_):
                 continue
             self.paragraphs[ALPHA[i]] += [int(v.strip()) for v in re_delims.split(line)]
 
-    def get_val(self, symbol):
-        if not symbol:
-            return 0
-        try:
-            return int(symbol)
-        except ValueError as e:
-            if re.search(r'[+&<>^_*/-]', symbol):
-                return self.expr_evaluate(symbol)
-            return self.variables.get(symbol, 0)
-
-    def assign(self, var, value):
-        v = self.get_val(value)
-        if DEBUG:
-            print('  ASSIGN "%s" = (%s) TO %s' % (value, v, var))
+    def assign(self, var, expr):
+        #temp = self.EXP
```

Comment:
looks like this saving and replacing of current expression buffer has no effect. Relying on previous values left in the buffer from non-adjacent items seems to be forbidden in this language.